middleware.go

```diff
@@ -33,9 +33,11 @@ func (r *Router) useInterface(mw middleware) {
 }
 
 // CORSMethodMiddleware sets the Access-Control-Allow-Methods response header
-// on a request, by matching routes based only on paths. It also handles
-// OPTIONS requests, by settings Access-Control-Allow-Methods, and then
-// returning without calling the next http handler.
+// on a request, by matching routes based only on paths.
+//
+// If a route in the router is set to handle OPTIONS calls, this middleware will
+// respond to OPTIONS calls with HTTP code "200 OK", the
+// Access-Control-Allow-Methods header and an empty body.
```

Comment:
The docs here should be clear that the middleware will respond with a default set of methods, and list those methods, unless a specific set of methods apply to that route.

However, the more I think on this, adding non-idempotent methods by default seems risky. Why not just safe + idempotent (yes, I'm aware that people break these semantics...) methods - `OPTIONS, GET, HEAD, TRACE` - as per https://tools.ietf.org/html/rfc7231#section-8.1.3 - ?